frontend/public/components/resource-dropdown.tsx

```diff
@@ -27,40 +26,14 @@ const blacklistResources = ImmutableSet([
   'events.k8s.io/v1beta1.Event',
 ]);
 
-const DropdownItem: React.SFC<DropdownItemProps> = ({ model, showGroup, checked }) => {
-  const { t } = useTranslation();
-  return (
-    <>
-      <span className={'co-resource-item'}>
-        <Checkbox
-          tabIndex={-1}
-          id={`${model.apiGroup}:${model.apiVersion}:${model.kind}`}
-          isChecked={checked}
-        />
-        <span className="co-resource-icon--fixed-width">
-          <ResourceIcon kind={referenceForModel(model)} />
-        </span>
-        <span className="co-resource-item__resource-name">
-          <span>
-            {model.labelKey ? t(model.labelKey) : model.kind}
-            {model.badge && model.badge === 'Tech Preview' && (
-              <span className="co-resource-item__tech-dev-preview">{t('public~Tech Preview')}</span>
-            )}
-          </span>
-          {showGroup && (
-            <span className="co-resource-item__resource-api text-muted co-truncate show co-nowrap small">
-              {model.apiGroup || 'core'}/{model.apiVersion}
-            </span>
-          )}
-        </span>
-      </span>
-    </>
-  );
-};
-
 const ResourceListDropdown_: React.SFC<ResourceListDropdownProps> = (props) => {
-  const { selected, onChange, allModels, className, groupToVersionMap } = props;
+  const { selected, onChange, allModels, groupToVersionMap, className } = props;
   const { t } = useTranslation();
+
+  const [isOpen, setIsOpen] = React.useState(false);
+  const [selectedOptions, setSelectedOptions] = React.useState(selected);
+  const [filterText, setFilterText] = React.useState(null);
```

Comment:
TypeScript won't be able to infer a type here since we're initializing the value to `null`, so it's better to declare it explicitly.

```suggestion
  const [filterText, setFilterText] = React.useState<string>(null);
```

Reply:
Added